src/main/java/com/avispl/symphony/dal/avdevices/bridges/vaddio/avbridgenano/VaddioBridgeNanoCommunicator.java

```diff
@@ -0,0 +1,1045 @@
+							logger.debug(String.format("PING DISCONNECTED: Connection to %s did not succeed within the timeout period of %sms", host, this.getPingTimeout()));
+						}
+						return this.getPingTimeout();
+					}
+				} catch (SocketTimeoutException | ConnectException tex) {
+					throw new SocketTimeoutException("Socket connection timed out");
+				} catch (UnknownHostException tex) {
+					throw new SocketTimeoutException("Socket connection timed out" + tex.getMessage());
+				} catch (Exception e) {
+					if (this.logger.isWarnEnabled()) {
+						this.logger.warn(String.format("PING TIMEOUT: Connection to %s did not succeed, UNKNOWN ERROR %s: ", host, e.getMessage()));
+					}
+					return this.getPingTimeout();
+				}
+			}
+			return Math.max(1, Math.toIntExact(pingResultTotal / this.getPingAttempts()));
+		} else {
+			throw new IllegalStateException("Cannot use device class without calling init() first");
+		}
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public List<Statistics> getMultipleStatistics() throws Exception {
+		ExtendedStatistics extendedStatistics = new ExtendedStatistics();
+		List<AdvancedControllableProperty> advancedControllableProperty = new ArrayList<>();
+		Map<String, String> stats = new HashMap<>();
+		Map<String, String> controlStats = new HashMap<>();
+		reentrantLock.lock();
+		try {
+			if (!isEmergencyDelivery) {
+				convertConfigManagement();
+				retrieveMonitoring();
+				if (localExtendedStatistics != null && localExtendedStatistics.getStatistics() == null && !isNextPollingInterval || !isConfigManagement) {
+					populateMonitoringAndControlling(stats, controlStats, advancedControllableProperty);
+					populateAudioInput(controlStats, advancedControllableProperty);
+					populateOutputControl(controlStats, advancedControllableProperty);
+					populateCrosspointGain(stats, controlStats, advancedControllableProperty);
+					if (isConfigManagement) {
+						stats.putAll(controlStats);
+						extendedStatistics.setControllableProperties(advancedControllableProperty);
+					}
+					extendedStatistics.setStatistics(stats);
+				}
+				if (localExtendedStatistics != null && localExtendedStatistics.getStatistics() != null && !localExtendedStatistics.getStatistics().isEmpty()) {
+					updateLocalExtendedStatisticsByPolingInterval(extendedStatistics, stats, controlStats, advancedControllableProperty);
+				}
+				localExtendedStatistics = extendedStatistics;
+			}
+			isEmergencyDelivery = false;
+		} finally {
+			reentrantLock.unlock();
+		}
+		return Collections.singletonList(localExtendedStatistics);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void controlProperties(List<ControllableProperty> list) throws Exception {
+		if (CollectionUtils.isEmpty(list)) {
+			throw new IllegalArgumentException("ControllableProperties can not be null or empty");
+		}
+		for (ControllableProperty p : list) {
+			try {
+				controlProperty(p);
+			} catch (Exception e) {
+				e.printStackTrace();
```

Comment:
I believe it was supposed to be a logger line here, instead of printStackTrace :)